self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryContracts.java

```diff
@@ -22,43 +22,378 @@
+            .map(Project::contracts)
+            .findFirst().orElse(new Contracts.Empty());
     }
 
     @Override
     public Iterator<Contract> iterator() {
-        return null;
+        return contracts.iterator();
+    }
+
+    /**
+     * Creates and adds a Contract based on a project id and contributor id.
+     *
+     * @param projectId Project's id
+     * @param contributorId Contributor's id
+     */
+    public void addContract(final int projectId, final int contributorId) {
+        final boolean contractExists = contracts.stream()
+            .anyMatch(c -> c.contributor().contributorId() == contributorId
+                && c.project().projectId() == projectId);
+        if (!contractExists) {
+            final ProjectInternal project = findOrCreateProject(projectId);
+            final ContributorInternal contributor =
+                findOrCreateContributor(contributorId);
+            final ContractInternal contract =
+                new ContractInternal(project, contributor);
+            //store the contract
+            contracts.add(contract);
+            //link this contract to its project and contributor
+            project.addContract(contract);
+            contributor.addContract(contract);
+        }
+    }
+
+    /**
+     * Returns a project by id from the projects set.
+     * If the project is not found, a project will be created and added to set.
+     *
+     * @param projectId Project's
+     * @return Internal representation of a Project
+     */
+    private ProjectInternal findOrCreateProject(final int projectId) {
+        ProjectInternal project = projects.stream()
+            .filter(p -> p.projectId() == projectId)
+            .findFirst()
+            .orElse(null);
+        if (project == null) {
+            project = new ProjectInternal(projectId, this);
+            projects.add(project);
+        }
+        return project;
+    }
+
+    /**
+     * Returns a contributor by id from the contributors set.
+     * If the contributor is not found, a contributor will
+     * be created and added to set.
+     *
+     * @param cbId Contributor's id
+     * @return Internal representation of a Contributor
+     */
+    private ContributorInternal findOrCreateContributor(final int cbId) {
+        ContributorInternal contributor = contributors.stream()
+            .filter(c -> c.contributorId() == cbId)
+            .findFirst()
+            .orElse(null);
+        if (contributor == null) {
+            contributor = new ContributorInternal(cbId, this);
+            contributors.add(contributor);
+        }
+        return contributor;
+    }
+
+
+    /**
+     * In memory implementation of a {@link Project}.
+     * <br>
+     * Aids {@link InMemoryContracts}
+     * <br>
+     * Has capability to link a contract.
+     * See: {@link ProjectInternal#addContract(Contract)}.
+     */
+    private static final class ProjectInternal implements Project {
```

Comment:
@criske I think ProjectInternal and ContributorInternal should be similar to the ``StoredProject`` and ``StoredProjectManager`` we have.